clicache/src/Cache.cpp

```diff
@@ -57,6 +56,21 @@ namespace Apache
         m_typeRegistry = gcnew Apache::Geode::Client::TypeRegistry(this);
       }
 
+      Cache::~Cache() { //Destructor - deterministic
+        if (!_disposed) {
+          //Clean-up managed resources
+          this->!Cache();
+          _disposed = true;
```

Comment:
Not thread safe. If two threads both invoke dispose, boom! Given the C++/CLI auto generates the `Dispose` method around the destructor we really aren't given a chance to make that discoverable in the documentation. 

Safest to use an atomic check and set operation in the if check.

Reply:
You can also just `lock` around the `if` block.

Reply:
Good catch.